packages/neso/package.py

```diff
@@ -167,4 +169,8 @@ def cmake_args(self):
                 args.append("-DACPP_TARGETS=cuda-nvcxx")
             args.append("-DSYCL_DEVICE_FILTER=GPU")
 
+        if "+libonly" in self.spec:
+            args.append("-DENABLE_NESO_SOLVERS=OFF")
+            args.append("-DENABLE_NESO_TESTS=OFF")
+
```

Comment:
I vote that we prefix all of the neso flags with NESO_ (especially since nektar and NP already follow that convention)

Personally, I'd go with `NESO_BUILD_SOLVERS` and `NESO_BUILD_TESTS` so that `ENABLE_` could be reserved for toggling functionality (ENABLE_X_OUTPUT, ENABLE_MODEL_Y etc.) rather than building/not building sections of the code.
Or we could go with `NESO_ENABLE_TESTS` and `NESO_ENABLE_SOLVERS` for consistency with NP... up to you